### PR TITLE
Enable returning individual samples 

### PR DESCRIPTION
**Pull request by LaurinFischer**
This PR implements functionality to optionally return the POVM measurements as individual samples rather than counts dictionaries, see #33. 

Essentially, the abstract method `_counter` of `POVMImplementation` is replaced by a new method `_povm_outcomes` that returns a list of samples instead of a counts dictionary. Similarly, the `get_counts_from_raw` method is renamed to `get_povm_outcomes_from_raw` and has a boolean flag that determines whether to return samples or counts. 
 
A small note on this has been added to the `getting_started.ipynb` tutorial.

### CODE REVIEW

File: povm_toolbox/library/povm_implementation.py [hunk 1]
```diff
@@ -174,34 +175,38 @@ def _get_bitarray(self, data: DataBin) -> BitArray:
...
 
     @abstractmethod
-    def _counter(
+    def _povm_outcomes(
         self,
         bit_array: BitArray,
         povm_metadata: MetadataT,
         loc: int | tuple[int, ...] | None = None,
-    ) -> Counter:
+    ) -> List:
```
Comment (mrossinek) [inline povm_toolbox/library/povm_implementation.py:183]:
```suggestion
    ) -> list[int]:
```
We are already using type hints using the builtin types.
Please double-check whether `int` is indeed the correct internal type.

File: povm_toolbox/library/povm_implementation.py [hunk 2]
```diff
@@ -174,34 +175,38 @@ def _get_bitarray(self, data: DataBin) -> BitArray:
...
         """TODO."""
 
-    def get_counts_from_raw(
+    def get_povm_outcomes_from_raw(
         self,
         data: DataBin,
         povm_metadata: MetadataT,
         loc: int | tuple[int, ...] | None = None,
+        return_counts = True
```
Comment (mrossinek) [inline povm_toolbox/library/povm_implementation.py:191]: Would you say that this is preferable to having two separate methods?
Say e.g. `get_povm_outcomes_from_raw` and `get_povm_counts_from_raw`?

**Review by mrossinek — COMMENTED** [review]
Overall this looks good. Some minor comments/suggestions.

I will revisit unittests and documentation of these methods in a future PR.

**Comment by mrossinek** [comment]
And to fix the failing CI, you can run `tox -e style` from the root of your local clone to try and fix this automatically. `tox -e lint` will tell you if it got everything.

File: povm_toolbox/library/povm_implementation.py (hunk 2 above)
Comment (LaurinFischer) [reply povm_toolbox/library/povm_implementation.py:191]: I'd say having one method saves on some code duplication. If there is no conceivable way to obtain the counts without first computing the list of individual samples and the calling `Counter` on it, I think having one method is a bit cleaner.

Comment (mrossinek) [reply povm_toolbox/library/povm_implementation.py:191]: Well, the second function could almost be as simply as this:
```python
def get_povm_counts_from_raw(...):
    return Counter(self.get_povm_outcomes_from_raw)
```
So I don't think that there is really a lot of code duplication.

The reason I bring this up, is that I believe having arguments which alternate between two different return types of a function is quite bad code design. I know that this is quite "common" in Qiskit(-related) projects, but that does not mean I have to agree with it and implement it here :smile:

Comment (LaurinFischer) [reply povm_toolbox/library/povm_implementation.py:191]: Alright, I'll make the change. 👍

Comment (LaurinFischer) [reply povm_toolbox/library/povm_implementation.py]: I've refactored this into two functions now. There is some duplication required to handle the different cases of the input arguments, and I've kept the original design that returns either an array or a Counter object (list of tuples for samples) depending on the input.

Comment (mrossinek) [reply povm_toolbox/library/povm_implementation.py]: Okay thanks, I will take a look :+1:

**Review by mrossinek — APPROVED** [review]
Yes, this looks good to me now :+1: 
Thanks a lot!

**Pull request merged**